pkg/errors/error.go

```diff
@@ -42,6 +42,12 @@ var (
 	// Terminal is returned with resource is in Terminal Condition
 	Terminal = fmt.Errorf(
 		"resource is in terminal condition")
+	// SecretTypeNotSupported is returned if non opaque secret is used.
+	SecretTypeNotSupported = fmt.Errorf(
+		"only opaque secrets can be used")
+	// SecretTypeNotFound is returned if secret does not exist.
```

Comment:
s/SecretTypeNotFound/SecretNotFound/

However, I don't think we really need a type-specific NotFound error in this package, do we? Can we just return the `NotFound` error?